Library/Homebrew/livecheck/strategy.rb

```diff
@@ -119,25 +120,68 @@ def self.page_headers(url)
 
       # Fetches the content at the URL and returns a hash containing the
       # content and, if there are any redirections, the final URL.
+      # If `curl` encounters an error, the hash will contain a `:messages`
+      # array with the error message instead.
       #
       # @param url [String] the URL of the content to check
       # @return [Hash]
       sig { params(url: String).returns(T::Hash[Symbol, T.untyped]) }
       def self.page_content(url)
         original_url = url
 
-        # Manually handling `URI#open` redirections allows us to detect the
-        # resolved URL while also supporting HTTPS to HTTP redirections (which
-        # are normally forbidden by `OpenURI`).
-        begin
-          content = URI.parse(url).open(redirect: false, &:read)
-        rescue OpenURI::HTTPRedirect => e
-          url = e.uri.to_s
-          retry
+        args = curl_args(
+          "--compressed",
+          # Include HTTP response headers in output, so we can identify the
+          # final URL after any redirections
+          "--include",
+          # Follow redirections to handle mirrors, relocations, etc.
+          "--location",
+          # cURL's default timeout can be up to two minutes, so we need to
+          # set our own timeout settings to avoid a lengthy wait
+          "--connect-timeout", "10",
+          "--max-time", "15"
+        )
+
+        stdout, stderr, status = curl_with_workarounds(
+          *args, url,
+          print_stdout: false, print_stderr: false,
+          debug: false, verbose: false,
+          user_agent: :default, timeout: 20,
```

Comment:
This differs only by `user_agent: :default` compared to the above. If this is changed: would we want it changed above? If so: can this be split into a method returning these arguments as an array or calling `curl_with_workarounds`?

Reply:
Originally this PR also added a `user_agent` parameter to `#page_content`, to lay the foundation for being able to configure the user agent in a `livecheck` block in the future. We decided it wasn't necessary for this particular PR and should be handled separately, so I removed it and left `user_agent: :default` as a placeholder.

Both `:default` and `nil` (no `user_agent` argument provided) use the same user agent string, so this wasn't doing anything meaningful other than visually aligning the corresponding parts of `#page_headers` and `#page_content`.

For the sake of simplicity, I've removed `user_agent: :default` and I'll add the option back in a later PR when I add the ability to set the `user_agent` in a `livecheck` block.